From a15f036b4e47dcbc8e9f4b739605975658ddb61f Mon Sep 17 00:00:00 2001
From: Alessandro Rubini <rubini@gnudd.com>
Date: Sat, 21 Jun 2014 08:48:04 +0200
Subject: [PATCH 11/12] sam9m10g45ek (for wrs): final partitions for V4.1

This changes the partitions in an incompatible way: then NAND
now has one partition for barebox environment (1M: 5 blocks to
protect against bad blocks) and one big partition fro UBI volumes.
Real stuff is then split in UBI volumes. Please see documentation
(in a later commit) for details.

Also, this adds the read-only hwinfo in dataflash, for V4.1 stuff.

Signed-off-by: Alessandro Rubini <rubini@gnudd.com>
---
 arch/arm/mach-at91/board-sam9m10g45ek.c |   19 ++++++++-----------
 1 file changed, 8 insertions(+), 11 deletions(-)

diff --git a/arch/arm/mach-at91/board-sam9m10g45ek.c b/arch/arm/mach-at91/board-sam9m10g45ek.c
index afc6418..552c209 100644
--- a/arch/arm/mach-at91/board-sam9m10g45ek.c
+++ b/arch/arm/mach-at91/board-sam9m10g45ek.c
@@ -26,6 +26,7 @@
 #include <linux/clk.h>
 #include <linux/atmel-mci.h>
 #include <linux/spi/flash.h>
+#include <linux/mtd/mtd.h>
 
 #include <mach/hardware.h>
 #include <video/atmel_lcdc.h>
@@ -106,6 +107,12 @@ static struct mtd_partition wrs_df_parts[] = {
 		.size	= 0x8400,
 	},
 	{
+		.name	= "hwinfo",
+		.offset	= MTDPART_OFS_APPEND,
+		.size	= 0x840,
+		.mask_flags = MTD_WRITEABLE,
+	},
+	{
 		.name	= "Available-dataflash",
 		.offset	= MTDPART_OFS_APPEND,
 		.size	= MTDPART_SIZ_FULL,
@@ -167,18 +174,8 @@ static struct mtd_partition __initdata ek_nand_partition[] = {
 		.size	= SZ_1M,
 	},
 	{
-		.name	= "Kernel",
+		.name	= "UBIfied-NAND",
 		.offset	= 1 << 20,
-		.size	= SZ_8M,
-	},
-	{
-		.name	= "Filesystem", /* We _want_ this to be mtd1 */
-		.offset	= 64 << 20,
-		.size	= SZ_128M,
-	},
-	{
-		.name	= "Available-nand",
-		.offset	= MTDPART_OFS_APPEND,
 		.size	= MTDPART_SIZ_FULL,
 	},
 };
-- 
1.7.10.4

